### PR TITLE
[AMORO-3256] Amoro Open Api with Swagger in Ams

### LINKED ISSUE #3256: [Improvement]:  Amoro Open Api with Swagger in Ams
### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/amoro/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

We need a clear documentation guide to tell how to use and call the API. We can refer to [polaris open api](https://polaris.apache.org/in-dev/unreleased/polaris-management-service/) approach and introduce Swagger to build corresponding clear documentation. 

### How should we improve?

Add Swagger in AMS Controller and build docs

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by mansonliwh**
## Why are the changes needed?

Close #3256.

## Brief change log

This PR introduces **Swagger** and **OpenAPI Generator**. To avoid intrusive development, this design does not use a large number of annotations on the controller. Instead, it predefines the **_openapi.yaml_**. The interface specifications defined in **_openapi.yaml_** are used to generate the Swagger page, API docs, and an SDK using the Apache HttpClient framework.

In the Maven build, you can use the generate-sdk profile to specify whether to generate the SDK.

## How was this patch tested?

- [ ☑️ ] Add screenshots for manual tests if appropriate

Swagger Page:
image

Auth Setting:
image

Generated API docs & SDK demo:
image

- [ ☑️ ] Run test locally before making a pull request

### CODE REVIEW

File: amoro-ams/pom.xml [hunk 1]
```diff
@@ -408,6 +409,66 @@
             <scope>test</scope>
         </dependency>
 
+        <dependency>
```
Comment (czy006) [inline amoro-ams/pom.xml:412]: we can use ${swagger.version} replace 2.2.10

File: amoro-ams/pom.xml [hunk 2]
```diff
@@ -503,6 +564,70 @@
                     <skip>true</skip>
                 </configuration>
             </plugin>
+
+            <plugin>
+                <groupId>org.openapitools</groupId>
+                <artifactId>openapi-generator-maven-plugin</artifactId>
```
Comment (czy006) [inline amoro-ams/pom.xml:570]: the latest version is [7.9.0](https://mvnrepository.com/artifact/org.openapitools/openapi-generator-maven-plugin/7.9.0) why we don't use it?

File: amoro-ams/pom.xml [hunk 3]
```diff
@@ -35,6 +35,7 @@
     <url>https://amoro.apache.org</url>
 
     <properties>
+        <skip-generate-sdk>false</skip-generate-sdk>
```
Comment (zhoujinsong) [inline amoro-ams/pom.xml:38]: IMO, we should not generate the SDK by default as it is not required for everyone.
But the doc is necessary for the dashboard.

So we may need to separate the properties for SDK and DOC.

File: amoro-ams/pom.xml (hunk 1 above)
Comment (zhoujinsong) [reply amoro-ams/pom.xml:412]: Yes, we may define the versions in properties.

File: amoro-ams/pom.xml [hunk 4]
```diff
@@ -408,6 +409,66 @@
             <scope>test</scope>
         </dependency>
 
+        <dependency>
+            <groupId>io.swagger.core.v3</groupId>
+            <artifactId>swagger-annotations</artifactId>
+            <version>2.2.10</version>
+        </dependency>
+
+        <dependency>
```
Comment (zhoujinsong) [inline amoro-ams/pom.xml:418]: Do we need the dependency by default?
Or should we separate them for the doc and the SDK?

File: amoro-ams/pom.xml [hunk 5]
```diff
@@ -408,6 +409,66 @@
...
+            <version>4.10.3</version>
+        </dependency>
+
+        <dependency>
+            <groupId>io.javalin</groupId>
+            <artifactId>javalin-openapi</artifactId>
+            <version>4.1.1</version>
+        </dependency>
+
+        <dependency>
```
Comment (zhoujinsong) [inline amoro-ams/pom.xml:454]: We already had httpclient:5.2 and httpcore:5.2 dependencies in runtime scope, do we still need this?

**Review by zhoujinsong — COMMENTED** [review]
@mansonliwh Thanks for the contribution! I left some comments.

File: amoro-web/src/components/SwaggerUI.vue [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+<script>
+import SwaggerUI from 'swagger-ui'
+import 'swagger-ui/dist/swagger-ui.css'
+
+export default {
+  name: 'SwaggerUI',
+  mounted() {
+    SwaggerUI({
+      dom_id: '#swagger-ui',
+      url: '/swagger-docs', // 这里是你的 OpenAPI 规范文件的路径
```
Comment (huyuanfeng2018) [inline amoro-web/src/components/SwaggerUI.vue:31]: Comments should be English.

File: amoro-web/src/router/index.ts [hunk 7]
```diff
@@ -101,6 +101,11 @@ const routes: Array<RouteRecordRaw> = [
     name: 'Login',
     component: Login,
   },
+  {
+    path: '/swagger-ui',
+    name: 'SwaggerUI',
+    component: SwaggerUI, // 使用 SwaggerUI 组件
```
Comment (huyuanfeng2018) [inline amoro-web/src/router/index.ts:107]: Here too.

**Review by czy006 — COMMENTED** [review]
At present, we have no relevant descriptive documents for this section. Maybe we need to supplement them.

**Review by czy006 — APPROVED** [review]
LGTM

File: amoro-ams/pom.xml [hunk 8]
```diff
@@ -35,6 +35,9 @@
     <url>https://amoro.apache.org</url>
 
     <properties>
+        <httpmime.version>4.5.13</httpmime.version>
+        <openapi.generator.version>6.0.0</openapi.generator.version>
```
Comment (zhoujinsong) [inline amoro-ams/pom.xml:39]: Is `openapi-generator-maven-plugin.version` a proper name?

File: amoro-ams/pom.xml [hunk 9]
```diff
@@ -35,6 +35,9 @@
...
+        <swagger.webjars.version>5.17.14</swagger.webjars.version>
```
Comment (zhoujinsong) [inline amoro-ams/pom.xml:40]: Is `swagger-ui.version` a proper name?

**Review by zhoujinsong — APPROVED** [review]
LGTM.

Thanks for the contribution!

**Pull request merged**